mslib/mscolab/server.py

```diff
@@ -351,6 +352,40 @@ def get_user():
     return json.dumps({'user': {'id': g.user.id, 'username': g.user.username}})
 
 
+@APP.route('/upload_profile_image', methods=["POST"])
+@verify_user
+def upload_profile_image():
+    user_id = request.form['user_id']
+    file = request.files['image']
+    if not file:
+        return jsonify({'message': 'No file provided or invalid file type'}), 400
+    if not file.mimetype.startswith('image/'):
+        return jsonify({'message': 'Invalid file type'}), 400
+    if file.content_length > mscolab_settings.MAX_UPLOAD_SIZE:
+        return jsonify({'message': 'File too large'}), 413
+
+    success, message = fm.save_user_profile_image(user_id, file)
+    if success:
+        return jsonify({'message': message}), 200
+    else:
+        return jsonify({'message': message}), 400
+
+
+@APP.route('/fetch_profile_image', methods=["GET"])
+@verify_user
+def fetch_profile_image():
+    user_id = request.form['user_id']
+    user = User.query.get(user_id)
+    if user and user.profile_image_path:
+        # base_directory = mscolab_settings.UPLOAD_FOLDER
+        # base_path = os.path.join(base_directory, 'profile')
+        base_path = mscolab_settings.PROFILE_IMG_FOLDER
+        filename = os.path.basename(user.profile_image_path)
+        return send_from_directory(base_path, filename)
```

Comment:
At this point the whole `fs` abstraction breaks down, because `send_from_directory` expects `os.path` semantics and we shouldn't really use anything else. Until here, everything should be treated as `fs`, and then can be somewhat cleanly converted to something `send_from_directory` understands with `send_from_directory(_fs.getsyspath(""), user.profile_image_path)` (`_fs` would need to be opened first).

Reply:
@workaryangupta 
Here a ToDo can be added, e.g. see open Discussion about PyFilesystem2 on ticket
or it can be solved as @matrss wrote.

Reply:
I disagree, this shouldn't be a todo, this is a bug that has already bitten us in other places. The fix is trivial, so it can be done here:
```python
        base_path = mscolab_settings.UPLOAD_FOLDER
        if sys.platform.startswith('win'):
            base_path = base_path.replace('\\', '/')
        filename = user.profile_image_path
        with fs.open_fs(base_path) as _fs:
            return send_from_directory(_fs.getsyspath(""), filename)
```